analytical_engine/apps/centrality/katz/katz_centrality.h

```diff
@@ -69,58 +69,55 @@ class KatzCentrality : public AppBase<FRAG_T, KatzCentralityContext<FRAG_T>>,
     return false;
   }
 
-  template <typename FRAG_T_, typename = void>
-  struct PullAndSend {
-    void operator()(const fragment_t& frag, context_t& ctx,
-                    message_manager_t& messages) {
-      auto inner_vertices = frag.InnerVertices();
-      auto& x = ctx.x;
-      auto& x_last = ctx.x_last;
-
-      for (auto& v : inner_vertices) {
-        auto es = frag.GetIncomingAdjList(v);
-        x[v] = 0;
-        for (auto& e : es) {
-          // do the multiplication y^T = Alpha * x^T A - Beta
-          x[v] += x_last[e.get_neighbor()];
+  /*
+    template <typename FRAG_T_, typename = void>
+    struct PullAndSend {
+      void operator()(const fragment_t& frag, context_t& ctx,
+                      message_manager_t& messages) {
+        auto inner_vertices = frag.InnerVertices();
+        auto& x = ctx.x;
+        auto& x_last = ctx.x_last;
+
+        for (auto& v : inner_vertices) {
+          auto es = frag.GetIncomingAdjList(v);
+          x[v] = 0;
+          for (auto& e : es) {
+            // do the multiplication y^T = Alpha * x^T A - Beta
+            x[v] += x_last[e.get_neighbor()];
+          }
+          x[v] = x[v] * ctx.alpha + ctx.beta;
+          messages.SendMsgThroughEdges(frag, v, ctx.x[v]);
         }
-        x[v] = x[v] * ctx.alpha + ctx.beta;
-        messages.SendMsgThroughEdges(frag, v, ctx.x[v]);
       }
-    }
-  };
-
-  template <typename FRAG_T_>
-  struct PullAndSend<
-      FRAG_T_, typename std::enable_if<!std::is_same<
-                   typename FRAG_T_::edata_t, grape::EmptyType>::value>::type> {
-    void operator()(const fragment_t& frag, context_t& ctx,
-                    message_manager_t& messages) {
-      auto inner_vertices = frag.InnerVertices();
-      auto& x = ctx.x;
-      auto& x_last = ctx.x_last;
-
-      for (auto& v : inner_vertices) {
-        auto es = frag.GetIncomingAdjList(v);
-        x[v] = 0;
-        for (auto& e : es) {
-          // do the multiplication y^T = Alpha * x^T A - Beta
-          double edata = 1.0;
-          static_if<!std::is_same<edata_t, grape::EmptyType>{}>(
-              [&](auto& e, auto& data) {
-                data = static_cast<double>(e.get_data());
-              })(e, edata);
-          x[v] += x_last[e.get_neighbor()] * edata;
-        }
-        x[v] = x[v] * ctx.alpha + ctx.beta;
-        messages.SendMsgThroughEdges(frag, v, ctx.x[v]);
+    };
+    */
```

Comment:
Delete the comment block if not used. If you want to keep the block for further debugging/profiling, you could use

```
#if 0
....
#else
...
#endif
```

Reply:
sorry, forget to remove the redundancy comments...